ui/src/components/ClusterCard.js

```diff
@@ -47,21 +70,124 @@ class ClusterCard extends React.Component {
     this.state = {
       clusterStatus: this.props.clusterModel.status,
     };
+    // This tracks internal state and should not be attached to "this.state" since
+    // updates to the component state will trigger rendering.
+    this.fastStatusRefreshes = 0
+  }
+
+  setClusterStatusDeleting = () => {
+    this.setState({ clusterStatus: "Deleting" });
+    this.checkForUpdatedState()
   }
 
-  setClusterStatusRunning = () => {
-    this.setState({ clusterStatus: "Running" });
+  /**
+   * Manage timing of cluster state re-fetches.
+   * This function will refresh the cluster status every ~5 minutes for running clusters
+   * and every ~30 seconds for clusters whose state is currently changing. For clusters
+   * that are "in flux", the refreshes will eventually give up and set the status to
+   * unknown.
+   */
+  checkForUpdatedState = () => {
+    // Default actions; do nothing.
+    var nextRefreshTimeout = 0;  // Timeout for next refresh cycle, 0 == don't refresh again.
+    var refreshFromService = false;  // Should we check the API this cycle?
+    var stateInFlux = (statusInFlux.indexOf(this.state.clusterStatus) >= 0);
+    // Use rapid status-updates up to 30 times when state.clusterStatus is in flux.
+    if (stateInFlux && this.fastStatusRefreshes < maxFastStatusChecks) {
+      nextRefreshTimeout = fastStatusCheckInterval
+      refreshFromService = true
+      this.fastStatusRefreshes += 1
+    // If the fast checks have been expended, set state to unknown and give up.
```

Comment:
Yes, manual refresh (in upper right corner) refreshes all cards via the "list clusters" request. Because the cluster card component is keyed by project/name/status (see [here](https://github.com/DataBiosphere/leonardo/blob/develop/ui/src/components/ClusterCardList.js#L23)) any update in status from the request will toss the old card component instance and replace it with a new instance with fresh state.